File: three_balls_ruby_kolpakov/balls.rb
require_relative('constants.rb')
require_relative('basket.rb')
require_relative('simulator.rb')

simulator = Simulator.new(1000000)
simulator.run
results = simulator.get_results

probability = results[WHITE][WHITE][RED][:count].to_f / results[WHITE][WHITE][:count].to_f

puts probability
puts

balls = [WHITE, BLACK, RED]

balls.each do |first|
    puts 'FIRST ' + first + ' ' + results[first][:count].to_s

    balls.each do |second|
        puts '    SECOND ' + second + ' ' + results[first][second][:count].to_s

        balls.each do |third|
            puts '        THIRD ' + third + ' ' + results[first][second][third][:count].to_s
        end
    end
end

File: two_balls_python_kolpakov/balls.py
from simulator import Simulator
import sys

number_of_tries = int(sys.argv[1])
simulator = Simulator(number_of_tries)
stats = simulator.getStats()
probabilty = stats['white_balls_on_the_first_step']['white_balls_on_the_second_step_count'] / stats['white_balls_on_the_first_step']['count']

print('TOTAL NUMBER OF TRIES: ' + str(number_of_tries) + '\n')
print('WHITE BALLS ON THE FIRST STEP: ' + str(stats['white_balls_on_the_first_step']['count']))
print('   Second step, white balls: ' + str(stats['white_balls_on_the_first_step']['white_balls_on_the_second_step_count']))
print('   Second step, black balls: ' + str(stats['white_balls_on_the_first_step']['black_balls_on_the_second_step_count']))
print()
print('BLACK BALLS ON THE FIRST STEP: ' + str(stats['black_balls_on_the_first_step']['count']))
print('   Second step, white balls: ' + str(stats['black_balls_on_the_first_step']['white_balls_on_the_second_step_count']))
print('   Second step, black balls: ' + str(stats['black_balls_on_the_first_step']['black_balls_on_the_second_step_count']))
print()
print('PROBABILITY: ' + str(probabilty))

